web/src/js/ducks/flows/index.ts

```diff
@@ -0,0 +1,375 @@
+        // Update highlight
+        if (matching_filters[FilterName.Highlight] === true) {
+            highlighted = new Set(highlighted);
+            highlighted.add(flow.id);
+        }
+
+        return {
+            list,
+            _listIndex,
+            byId,
+            view,
+            _viewIndex,
+            sort,
+            selected,
+            selectedIds,
+            highlighted,
+        };
+    } else if (FLOWS_UPDATE.match(action)) {
+        const { flow, matching_filters } = action.payload;
+        const { _listIndex, sort, selectedIds } = state;
+        let { view, _viewIndex, selected, highlighted } = state;
+        // Update list
+        const listPos = state._listIndex.get(flow.id)!;
+        const list = [...state.list];
+        list[listPos] = flow;
+        const byId = new Map(state.byId);
+        byId.set(flow.id, flow);
+        // Update view
+        const prevViewPos: number | undefined = _viewIndex.get(flow.id);
+        const hasOldFlow = prevViewPos !== undefined;
+        const hasNewFlow =
+            matching_filters[FilterName.Search] === true ||
+            matching_filters[FilterName.Search] === undefined;
+        if (hasNewFlow && !hasOldFlow) {
+            ({ view, _viewIndex } = insertViewItem(
+                view,
+                _viewIndex,
+                flow,
+                makeSort(sort),
+            ));
+        } else if (!hasNewFlow && hasOldFlow) {
+            ({ view, _viewIndex } = removeViewItemAt(
+                view,
+                _viewIndex,
+                prevViewPos,
+            ));
+        } else if (hasNewFlow && hasOldFlow) {
+            ({ view, _viewIndex } = updateViewItem(
+                view,
+                _viewIndex,
+                flow,
+                makeSort(sort),
+            ));
+        }
+        // Update selection
+        if (selectedIds.has(flow.id)) {
+            selected = selected.map((existing) =>
+                existing.id === flow.id ? flow : existing,
+            );
+        }
+        // Update highlight
+        if (matching_filters[FilterName.Highlight]) {
+            if (!highlighted.has(flow.id)) {
+                highlighted = new Set(highlighted);
+                highlighted.add(flow.id);
+            }
+        } else {
+            highlighted = withElemRemoved(highlighted, flow.id);
+        }
+
+        return {
+            list,
+            _listIndex,
+            byId,
+            view,
+            _viewIndex,
+            sort,
+            selected,
+            selectedIds,
+            highlighted,
+        };
+    } else if (FLOWS_REMOVE.match(action)) {
+        const flow_id = action.payload;
+        const { sort } = state;
+        let { view, _viewIndex, selected, selectedIds } = state;
+        const listPos = state._listIndex.get(flow_id)!;
+        const list = state.list.toSpliced(listPos, 1);
+        const _listIndex = withKeyRemoved(state._listIndex, flow_id);
```

Comment:
There's still something not working as expected. When we have only one element and remove it, `list` doesn't become `[]`

Reply:
How do you remove the last flow? Seems to work for me.

Reply:
This happens if we previously had more than one flow. For example, if we add 3 flows and then remove them one by one from the bottom up, when removing the last one (at the top), `list` still contains that top flow.

Reply:
https://github.com/user-attachments/assets/4c0a1602-f035-41dd-a9d3-d9e40a0e284f